src/pages/AddLiquidity/APRCalculation.tsx

```diff
@@ -5,14 +5,29 @@ import BN from 'bignumber.js'
 export const apr = (
   lpPair: Pair,
   totalSupplyUni: BigNumber,
-  rewardRate: BigNumber | undefined,
-  totalStake: BigNumber | undefined
+  adminRate: BigNumber | undefined,
+  totalStake: BigNumber | undefined,
+  contractStage: BigNumber | undefined,
+  currentBlockNum: BigNumber | undefined
 ): number | undefined => {
-  if (rewardRate === undefined || totalStake === undefined) return undefined
-  const rewardRateBN = new BN(rewardRate.toString())
+  let APR: number
+  const endColBlock = new BN(15389352)
+  if (
+    adminRate === undefined ||
+    totalStake === undefined ||
+    contractStage === undefined ||
+    currentBlockNum === undefined
+  )
+    return undefined
+  if (contractStage.toString() === '3') return (APR = 0)
+  let rewardRateBN = new BN(adminRate.toString())
   const totalStakeBN = new BN(totalStake.toString())
   const totalSupplyUniBN = new BN(totalSupplyUni.toString())
   const blocksPerWeek = 6000 * 7
+  const currEpoch = new BN(currentBlockNum.toString()).minus(endColBlock).div(blocksPerWeek)
```

Comment:
currEpoch should be an integer, not a floating point number. 
Reason: if currEpoch = 2.5, then the loop below runs for 3 times, instead of 2 times as expected.
Therefore, we should use type @ethersproject/bignumber which is only used for integers